### PR TITLE
Add SignalRW helpers

### LINKED ISSUE #139: Make _RBV helper externally visible
The pattern of having `_RBV` at the end of a PV for the corresponding readback is common outside just AD. `ophyd-async` should provide a helper function for it. One already exists at https://github.com/bluesky/ophyd-async/blob/c42afcb1f801d5f3f6ae88fd6a7fa8e0c7043893/src/ophyd_async/epics/areadetector/utils.py#L11 but it feels weird for my device to depend on area detector just for this.

### PR DESCRIPTION
**Pull request by GDYendell**
Fixes #139

### CODE REVIEW

File: src/ophyd_async/epics/signal/signal.py
Comment (coretl) [inline src/ophyd_async/epics/signal/signal.py:1]: I might have got the wrong end of the stick, but I thought we were keeping `epics_signal_rw` as it is, and adding `epics_signal_rw_rbv(datatype, write_pv, read_suffix="_RBV")`...

@DominicOram what did you think?

Comment (olliesilvester) [reply src/ophyd_async/epics/signal/signal.py:1]: Yeah that's what is currently in dodal, I like it the most

Comment (callumforrester) [reply src/ophyd_async/epics/signal/signal.py:1]: I agree, would prefer a single function with a default argument

Comment (GDYendell) [reply src/ophyd_async/epics/signal/signal.py:1]: Updated to just add a new function that allows overriding the readback suffix.

File: src/ophyd_async/epics/signal/signal.py [hunk 1]
```diff
@@ -58,6 +58,28 @@ def epics_signal_rw(
     return SignalRW(backend)
 
 
+def epics_signal_rw_rbv(
```
Comment (callumforrester) [inline src/ophyd_async/epics/signal/signal.py:61]: I think we still want it to be called `epics_signal_rw`

Comment (coretl) [reply src/ophyd_async/epics/signal/signal.py:61]: nope, this is the additional function to specifically do `prefix`, `prefix + "_RBV"` PV pairs

File: src/ophyd_async/epics/signal/signal.py [hunk 2]
```diff
@@ -58,6 +58,28 @@ def epics_signal_rw(
...
+    datatype: Type[T],
+    read_pv: str,
+    write_pv: Optional[str] = None,
+    read_suffix: str = "_RBV",
```
Comment (coretl) [inline src/ophyd_async/epics/signal/signal.py:65]: We never need to provide a separate read pv, so suggest we change signature to:
```suggestion
    write_pv: str,
    read_suffix: str = "_RBV",
```

File: src/ophyd_async/epics/signal/signal.py [hunk 3]
```diff
@@ -58,6 +58,28 @@ def epics_signal_rw(
...
+    datatype:
+        Check that the PV is of this type
+    read_pv:
+        The PV to read and monitor
+    write_pv:
+        If given, use this PV to write to, otherwise use read_pv
+    read_suffix:
+        Append this suffix to the readback pv
+    """
+    return epics_signal_rw(datatype, f"{read_pv}{read_suffix}", write_pv or read_pv)
```
Comment (coretl) [inline src/ophyd_async/epics/signal/signal.py:80]: so this becomes
```suggestion
    return epics_signal_rw(datatype, f"{write_pv}{read_suffix}", write_pv)
```

**Review by coretl — CHANGES_REQUESTED** [review]

File: src/ophyd_async/epics/signal/signal.py (hunk 1 above)
Comment (coretl) [reply src/ophyd_async/epics/signal/signal.py:61]: nope, we already have one of these, this is a second function for when we have a `prefix`, `prefix + "_RBV"` record pair

File: src/ophyd_async/epics/signal/signal.py
Comment (DominicOram) [reply src/ophyd_async/epics/signal/signal.py:1]: Yes, new function, behaviour of `epics_signal_rw` need not change. This is literally just a small helper function to tidy up some devices here and there

File: src/ophyd_async/epics/signal/signal.py (hunk 2 above)
Comment (GDYendell) [reply src/ophyd_async/epics/signal/signal.py]: Done

File: src/ophyd_async/epics/signal/signal.py (hunk 3 above)
Comment (GDYendell) [reply src/ophyd_async/epics/signal/signal.py]: Done

File: src/ophyd_async/epics/signal/signal.py [hunk 4]
```diff
@@ -58,6 +58,25 @@ def epics_signal_rw(
     return SignalRW(backend)
 
 
+def epics_signal_rw_rbv(
+    datatype: Type[T],
+    write_pv: Optional[str] = None,
```
Comment (coretl) [inline src/ophyd_async/epics/signal/signal.py:63]:
```suggestion
    write_pv: str,
```

File: tests/core/test_signal.py
[the remark this answers is not in the document]
Comment (GDYendell) [reply tests/core/test_signal.py:1]: Done

**Review by coretl — APPROVED** [review]

**Comment by DominicOram** [comment]
Does it make sense to update the signals in AD with this e.g. https://github.com/bluesky/ophyd-async/blob/c42afcb1f801d5f3f6ae88fd6a7fa8e0c7043893/src/ophyd_async/epics/areadetector/utils.py#L11?

**Pull request merged**